### PR TITLE
Support smlsharp build

### PR DESCRIPTION
**Pull request by KeenS**
Most parts of works are adding smi files and adding smlsharp task to Makefile.
Also some fixes are made (though I thing install.sml is'nt used).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,35 +1,60 @@
 BIN=bin
+TARGET=$(BIN)/smackage
+
+MLTON=mlton
```
Comment (gian) [inline Makefile]: Thanks, this is a nice tidy-up!

**Comment by gian** [comment]
I'm a little confused. Does smlsharp require that all source files have a `.smi` suffix? If so, I'd prefer to selectively rename or copy the SML files programmatically during the build step (if no changes are required), or hide smlsharp-specific files in a subdirectory of `src`. Could we make either of those options work, somehow?

**Comment by gian** [comment]
Oh, I sorta answered my own question by RTFMing. Are the `smi` files required to exist within the same source path? This seems an awkward construction to have to support long-term.

**Comment by KeenS** [comment]
> Are the smi files required to exist within the same source path?

Yes. This is needed because SML# supports separate compilation. This the same way as OCaml does.

> This seems an awkward construction to have to support long-term.

I admit this may increase maintenance cost. If you don't think taking that cost to support extra implementation pays, don't hesitate to reject this patch. BTW interface files help code reader by showing file dependencies and structure signatures so interface files aren't mere cost.

**Comment by gian** [comment]
Yeah, this seems low-risk, mostly because (at worst) we break SML# support, and given that development isn't terribly active, that doesn't seem likely in the short term. 

Also, presumably we can revert by doing `rm *.smi` :p 

I'll merge and cut a new tagged version tomorrow. Thanks so much for contributing this, @KeenS!

**Comment by KeenS** [comment]
:beers:

**Comment by gian** [comment]
@robsimmons FYI ping tag etc etc

**Comment by gian** [comment]
@KeenS Done, and tagged as 1.4.0. Please let me know if you encounter issues!

**Comment by robsimmons** [comment]
Pong - sgtm

**Pull request merged**